bundles/net.tourbook/src/net/tourbook/tour/TourInfoUI.java

```diff
@@ -869,6 +871,15 @@ private void createUI_44_Weather(final Composite parent) {
 
       createUI_Label(parent, UI.UNIT_LABEL_TEMPERATURE);
 
+      /*
+       * Temperature from device
+       */
+      createUI_Label(parent, Messages.Tour_Tooltip_Label_Temperature_FromDevice);
+
+      _lblTemperature_FromDevice = createUI_LabelValue(parent, SWT.TRAIL);
+
+      createUI_Label(parent, UI.UNIT_LABEL_TEMPERATURE);
+
```

Comment:
I will improve the UI based on your suggestions

Reply:
>When possible, then the manual/provider temperature should only be visible when these values are set, maybe also for the device temperature but this may needs more changes

That's hard because by default, the temperature value is 0 which is a possible value in real life. So how can we know that those values are "set" ??

Reply:
Tooltip improved. Also, now the average temperature from the device is only displayed if the `temperatureSerie` is not empty

![image](https://user-images.githubusercontent.com/25235665/156966044-3026c7f8-23cc-4273-aa79-a0164c5242ea.png)

Reply:
Almost perfect, the weather values are looking a bit chaotic (similar like the weather sometimes is), I will try to improve it